test/unit_SUITE.erl

```diff
@@ -135,26 +135,32 @@ check_parse_arguments(ExpRes, Fun, As) ->
 
     true = SortRes(ExpRes) =:= SortRes(Fun(As)).
 
-filtering_flags_parsing(_Config) ->
-    Cases = [{[], [], []}
-            ,{[{"--online", true}], ["--offline", "--online", "--third-option"], [false, true, false]}
-            ,{[{"--online", true}, {"--third-option", true}, {"--offline", true}], ["--offline", "--online", "--third-option"], [true, true, true]}
-            ,{[], ["--offline", "--online", "--third-option"], [true, true, true]}
-            ],
-    lists:foreach(fun({Vals, Opts, Expect}) ->
-                          case rabbit_cli:filter_opts(Vals, Opts) of
-                              Expect ->
+mutually_exclusive_flags_parsing(_Config) ->
+    Spec = [{"--online", online}
+           ,{"--offline", offline}
+           ,{"--local", local}],
+    Default = all,
+    Cases =[{["--online"], {ok, online}}
+           ,{[], {ok, Default}}
+           ,{["--offline"], {ok, offline}}
+           ,{["--local"], {ok, local}}
+           ,{["--offline", "--local"], {error, "Options '--offline' and '--local' are mutually exclusive"}}
```

Comment:
These error checks are too closely tied to the implementation by verifying the exact message. For long term maintenance, it would be better to match only the `error` part in the tuple.